daemon/reload.go

```diff
@@ -31,15 +31,14 @@ func (tx *reloadTxn) OnRollback(cb func() error) {
 	tx.onCommit = append(tx.onRollback, cb)
 }
 
-func (tx *reloadTxn) run(cbs []func() error) error {
+func (tx *reloadTxn) run(cbs []func() error) (err error) {
 	tx.onCommit = nil
 	tx.onRollback = nil
 
-	var res *multierror.Error
 	for _, cb := range cbs {
-		res = multierror.Append(res, cb())
+		err = errors.Join(err, cb())
 	}
```

Comment:
I always get confused as "`errors.Join` in a loop" != `[]errs{}` -> `errors.Join(errs...)`, but I think in this case, the equivalent _may be_ to collect in a slice first;


```go
var errs []error
for _, cb := range cbs {
    errs = append(errs, cb())
}
return errors.Join(errs...)
```